packages/react-components/src/shared/src/augmentElement.tsx

```diff
@@ -0,0 +1,37 @@
+import { isString } from "lodash";
+import { isValidElementType } from "react-is";
+import { mergeProps } from "./mergeProps";
+import { normalizeSize } from "./normalizeSize";
+import React, { ElementType, HTMLAttributes, ReactElement, RefAttributes, cloneElement } from "react";
+import type { SizeAdapter } from "./createSizeAdapter";
+
+export function augmentElement(element: ReactElement & RefAttributes<any>, newProps: Record<string, any>) {
+    const augmentedProps = mergeProps({ ...element.props, ref: element.ref }, newProps);
+
+    return cloneElement(element, augmentedProps);
+}
+
+export function createOrAugmentElement<T extends string, P extends HTMLAttributes<T>>(element: ReactElement<P, T> | ElementType, props: P) {
+    if (isValidElementType(element) && !isString(element)) {
+        const Type = element;
+
+        return <Type {...props} />;
+    }
+
+    // The isString in the previous condition assume it can be possible that element is a string. however, if its the case, cloneElement does not support string, so it does not make sense
+    // Therefore, since we are only typing the code and i don't want to introduce regression bugs, i will disable the ts compiler for that line
```

Comment:
Yeah it happens because some components pass children directly to this function and children can be a string.

However, this function is deprecated it should be delete before the release.